src/shapirowilk.jl

```diff
@@ -0,0 +1,244 @@
+               (:C2, [0.0, 0.042981, -0.293762, -1.752461, 5.682633, -3.582633]),
+               (:C3, [0.5440, -0.39978, 0.025054, -0.0006714]),
+               (:C4, [1.3822, -0.77857, 0.062767, -0.0020322]),
+               (:C5, [-1.5861, -0.31082, -0.083751, 0.0038915]),
+               (:C6, [-0.4803, -0.082676, 0.0030302]),
+               (:C7, [0.164, 0.533]),
+               (:C8, [0.1736, 0.315]),
+               (:C9, [0.256, -0.00635]),
+               (:G, [-2.273, 0.459])]
+    @eval $(Symbol(:__RS92_, s))(x) = Base.Math.@horner(x, $(c...))
+end
+
+struct ShapiroWilkCoefs <: AbstractVector{Float64}
+    N::Int
+    A::Vector{Float64}
+end
+
+Base.size(SWc::ShapiroWilkCoefs) = (SWc.N,)
+Base.IndexStyle(::Type{ShapiroWilkCoefs}) = IndexLinear()
+
+Base.@propagate_inbounds function Base.getindex(SWc::ShapiroWilkCoefs, i::Integer)
+    @boundscheck checkbounds(SWc, i)
+    @inbounds if checkbounds(Bool, SWc.A, i)
+        return SWc.A[i]
+    elseif isodd(length(SWc)) && i == lastindex(SWc.A) + 1
+        return zero(eltype(SWc))
+    else
+        return -SWc.A[SWc.N + 1 - i]
+    end
+end
+
+function ShapiroWilkCoefs(N::Integer)
+    if N < 3
+        throw(ArgumentError("N must be greater than or equal to 3, got $N"))
+    elseif N == 3 # exact
+        return ShapiroWilkCoefs(N, [sqrt(2.0) / 2.0])
+    else
+        # Weisberg&Bingham 1975 statistic; store only positive half of m:
+        # it is (anti-)symmetric; hence '2' factor below
+        m = [-quantile(Normal(), (i - 3 / 8) / (N + 1 / 4)) for i in 1:div(N, 2)]
+        mᵀm = 2sum(abs2, m)
+        x = 1 / sqrt(N)
+        a₁ = m[1] / sqrt(mᵀm) + __RS92_C1(x) # aₙ = cₙ + (...)
+        if N ≤ 5
+            ϕ = (mᵀm - 2m[1]^2) / (1 - 2a₁^2)
+            m .= m ./ sqrt(ϕ) # A, but reusing m to save allocs
+            m[1] = a₁
+        else
+            a₂ = m[2] / sqrt(mᵀm) + __RS92_C2(x) # aₙ₋₁ = cₙ₋₁ + (...)
+            ϕ = (mᵀm - 2m[1]^2 - 2m[2]^2) / (1 - 2a₁^2 - 2a₂^2)
+            m .= m ./ sqrt(ϕ) # A, but reusing m to save allocs
+            m[1], m[2] = a₁, a₂
+        end
+        return ShapiroWilkCoefs(N, m)
+    end
+end
+
+function swstat(X::AbstractVector{<:Real}, A::ShapiroWilkCoefs)
+    if last(X) - first(X) < length(X) * eps()
+        throw(ArgumentError("sample is constant (up to numerical accuracy)"))
+    end
+    AX = dot(view(A, 1:length(X)), X)
+    m = mean(X)
+    S² = sum(x -> abs2(x - m), X)
+    W = AX^2 / S²
+    return min(W, one(W)) # to guard against numeric errors
+end
+
+struct ShapiroWilkTest <: HypothesisTest
+    coefs::ShapiroWilkCoefs # Expectation of order statistics for Shapiro-Wilk test
+    W::Float64              # test statistic
+    censored::Int   # only the smallest N₁ samples were used
+end
+
+testname(::ShapiroWilkTest) = "Shapiro-Wilk normality test"
+function population_param_of_interest(t::ShapiroWilkTest)
+    return ("Squared correlation of data and the expected order statistics of N(0,1) (W)",
```

Comment:
To save a few characters:
```suggestion
    return ("Squared correlation of data and expected order statistics of N(0,1) (W)",
```